### ISSUE TITLE
sdf_* functions not taking pipeline objects as inputs

### ISSUE THREAD

**Issue by benmwhite** [rstudio/sparklyr#1662]
Certain sdf_* functions do not work as expected within ML pipelines. I've tested this with a handful of the functions, but it may extend to others.

Using sparklyr 0.8.4 with R 3.4.1 on an YARN-managed cluster and R 3.5.1 on my laptop with a local Spark connection, but the error message is the same in either case.

According to the 0.8.4 documentation these functions can take `ml_pipeline` objects as inputs under the `x` argument, but instead they produce an error with the message 

>Error: Unable to retrieve a Spark DataFrame from object of class ml_pipeline ml_estimator ml_pipeline_stage

 when included. 

Reproducible examples:
```r
spark_version <- "2.3.0"
sc <- spark_connect(master = "local", version = spark_version)
```

Verifying the class:
```r
ml_pipeline(sc) %>%
  class()
```
This returns
>[1] "ml_pipeline"       "ml_estimator"     
[3] "ml_pipeline_stage"

The error occurs each of these calls:

```r
ml_pipeline(sc) %>%
  sdf_coalesce(1)

ml_pipeline(sc) %>%
  sdf_repartition()

ml_pipeline(sc) %>%
  sdf_sample()
```
In each of these cases I'm expecting the output to show the printout of the pipeline stages, like with the following: 

```r
ml_pipeline(sc) %>%
  ft_string_indexer("foo", "bar")
```

which returns

>Pipeline (Estimator) with 1 stage
<pipeline_71cc77b11a17> 
  Stages 
  |--1 StringIndexer (Estimator)
  |    <string_indexer_71cc2f732618> 
  |     (Parameters -- Column Names)
  |      input_col: foo
  |      output_col: bar
  |     (Parameters)
  |      handle_invalid: error
  |      string_order_type: frequencyDesc

Instead it shows the error message mentioned above.

**Comment by kevinykuo** [comment]
Right, these aren't supposed to work since they're not pipeline stages. Will need to update documentation.

**Comment by benmwhite** [comment]
Ah ok, thanks. So the ml_* and ft_* functions are the ones that can work for either pipelines or tables and the sdf_* functions are for table inputs only (i.e. coercible to Spark Data Frame), is that correct?

**Comment by kevinykuo** [comment]
Yep that's the spirit. The behavior of `ml_*` and `ft_*` functions changes according to what the first argument is, see the `Value` section of `?ml_logistic_regression()` as an example. `ml_predict/ml_transform()` are supposed to work with ml model or pipeline objects as the first argument, and `sdf_*` all require the first argument to be a `tbl_spark`.
